File: modules/apps/object/object-admin-rest-client-js/src/models/ObjectValidationRule.ts
/**
 * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

			import {ObjectValidationRuleSetting} from './ObjectValidationRuleSetting';

/**
 * @author Javier Gamarra
 * @generated
 */

	export class ObjectValidationRule {
			"actions"?: {[key: string]: {[key: string]: string;};};
			"active"?: boolean;
			"dateCreated"?: Date;
			"dateModified"?: Date;
			"engine"?: string;
			"engineLabel"?: string;
			"errorLabel"?: {[key: string]: string;};
			"externalReferenceCode"?: string;
			"id"?: number;
			"name"?: {[key: string]: string;};
			"objectDefinitionExternalReferenceCode"?: string;
			"objectDefinitionId"?: number;
			"objectValidationRuleSettings"?: Array<ObjectValidationRuleSetting>;
			"outputType"?: 'fullValidation' | 'partialValidation';
			"script"?: string;
			"system"?: boolean;

		static "discriminator": string | undefined = undefined;

	static "attributeTypeMap": Array<{
		baseName: string;
		name: string;
		type: string;
	}> = [
		{
			baseName: "actions",
			name: "actions",
			type: "{[key: string]: {[key: string]: string;};}",
		},
		{
			baseName: "active",
			name: "active",
			type: "boolean",
		},
		{
			baseName: "dateCreated",
			name: "dateCreated",
			type: "Date",
		},
		{
			baseName: "dateModified",
			name: "dateModified",
			type: "Date",
		},
		{
			baseName: "engine",
			name: "engine",
			type: "string",
		},
		{
			baseName: "engineLabel",
			name: "engineLabel",
			type: "string",
		},
		{
			baseName: "errorLabel",
			name: "errorLabel",
			type: "{[key: string]: string;}",
		},
		{
			baseName: "externalReferenceCode",
			name: "externalReferenceCode",
			type: "string",
		},
		{
			baseName: "id",
			name: "id",
			type: "number",
		},
		{
			baseName: "name",
			name: "name",
			type: "{[key: string]: string;}",
		},
		{
			baseName: "objectDefinitionExternalReferenceCode",
			name: "objectDefinitionExternalReferenceCode",
			type: "string",
		},
		{
			baseName: "objectDefinitionId",
			name: "objectDefinitionId",
			type: "number",
		},
		{
			baseName: "objectValidationRuleSettings",
			name: "objectValidationRuleSettings",
			type: "Array<ObjectValidationRuleSetting>",
		},
		{
			baseName: "outputType",
			name: "outputType",
			type: "'fullValidation' | 'partialValidation'",
		},
		{
			baseName: "script",
			name: "script",
			type: "string",
		},
		{
			baseName: "system",
			name: "system",
			type: "boolean",
		},
		];

		static getAttributeTypeMap() {
				return ObjectValidationRule.attributeTypeMap;
		}
	}

File: modules/apps/object/object-api/src/main/java/com/liferay/object/model/ObjectValidationRule.java
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.object.model;

import com.liferay.portal.kernel.annotation.ImplementationClassName;
import com.liferay.portal.kernel.model.PersistedModel;
import com.liferay.portal.kernel.util.Accessor;

import org.osgi.annotation.versioning.ProviderType;

/**
 * The extended model interface for the ObjectValidationRule service. Represents a row in the &quot;ObjectValidationRule&quot; database table, with each column mapped to a property of this class.
 *
 * @author Marco Leo
 * @see ObjectValidationRuleModel
 * @generated
 */
@ImplementationClassName(
	"com.liferay.object.model.impl.ObjectValidationRuleImpl"
)
@ProviderType
public interface ObjectValidationRule
	extends ObjectValidationRuleModel, PersistedModel {

	/*
	 * NOTE FOR DEVELOPERS:
	 *
	 * Never modify this interface directly. Add methods to <code>com.liferay.object.model.impl.ObjectValidationRuleImpl</code> and rerun ServiceBuilder to automatically copy the method declarations to this interface.
	 */
	public static final Accessor<ObjectValidationRule, Long>
		OBJECT_VALIDATION_RULE_ID_ACCESSOR =
			new Accessor<ObjectValidationRule, Long>() {

				@Override
				public Long get(ObjectValidationRule objectValidationRule) {
					return objectValidationRule.getObjectValidationRuleId();
				}

				@Override
				public Class<Long> getAttributeClass() {
					return Long.class;
				}

				@Override
				public Class<ObjectValidationRule> getTypeClass() {
					return ObjectValidationRule.class;
				}

			};

	public boolean compareOutputType(String outputType);

	public java.util.List<ObjectValidationRuleSetting>
		getObjectValidationRuleSettings();

	public void setObjectValidationRuleSettings(
		java.util.List<ObjectValidationRuleSetting>
			objectValidationRuleSettings);

}
// SB-Hash:-1993975747